js/exporter/jspdf/row_utils.js

```diff
@@ -0,0 +1,101 @@
+import { isDefined } from '../../core/utils/type';
+import { calculateRowHeight } from './pdf_utils';
+
+function getRows(doc, dataProvider, dataGrid, options) {
+    const rows = [];
+
+    let previousRow;
+    let currentRow;
+    const rowsCount = dataProvider.getRowsCount();
+    const wordWrapEnabled = !!dataGrid.option('wordWrapEnabled');
+
+    for(let rowIndex = 0; rowIndex < rowsCount; rowIndex++) {
+        previousRow = currentRow;
+        currentRow = createRow(dataProvider, rowIndex, previousRow, wordWrapEnabled);
+
+        if(options.customizeCell) {
+            currentRow.cells.forEach(cellInfo => {
+                options.customizeCell({ gridCell: cellInfo.gridCell, pdfCell: cellInfo.pdfCell });
+            });
+        }
+
+        rows.push(currentRow);
+    }
+
+    return rows;
+}
+
+function createRow(dataProvider, rowIndex, previousRow, wordWrapEnabled) {
+    const rowType = dataProvider.getCellData(rowIndex, 0, true).cellSourceData.rowType;
+    let indentLevel = rowType !== 'header' ? dataProvider.getGroupLevel(rowIndex) : 0;
+    if(rowType === 'groupFooter' && previousRow?.rowType === 'groupFooter') {
+        indentLevel = previousRow.indentLevel - 1;
```

Comment:
'indentLevel' is an option to create a 'row' description and it's better to calculate it before function is called and pass the resulting value as argument

Reply:
Yep, great idea - fixed